src/query/service/src/interpreters/interpreter_view_create.rs

```diff
@@ -69,7 +70,28 @@ impl CreateViewInterpreter {
     async fn create_view(&self) -> Result<PipelineBuildResult> {
         let catalog = self.ctx.get_catalog(&self.plan.catalog)?;
         let mut options = BTreeMap::new();
-        options.insert("query".to_string(), self.plan.subquery.clone());
+
+        let subquery = if self.plan.column_names.is_empty() {
+            self.plan.subquery.clone()
+        } else {
+            let mut planner = Planner::new(self.ctx.clone());
+            let (plan, _, _) = planner.plan_sql(&self.plan.subquery.clone()).await?;
+            if plan.schema().fields().len() != self.plan.column_names.len() {
+                return Err(ErrorCode::BadDataArrayLength(format!(
```

Comment:
Please advice if `BadDataArrayLength` is appropriate, thanks!

Reply:
It's ok